docs/sources/write/style-guide/write-for-developers/index.md

```diff
@@ -78,23 +78,135 @@ Additional tips:
 Readers of documentation typically skim through it to find code samples they can copy and paste and run as is.
 Because of this and whenever possible, provide production-ready examples.
 
-However, it isn't necessary for every code example to be runnable in production. Some code examples are written to illustrate a point so that the developer can learn how to do something similar on their own.
+However, it isn't necessary for every code example to be runnable in production.
+Some code examples are written to illustrate a point so that the developer can learn how to do something similar on their own.
+Such examples should be clearly marked as partial.
 
 When providing an example, give a written description. You can put it either in the body of the document or as explanatory comments within the example code.
 
 Remember the basic rule: explain _why_ your code does what it does, rather than describe _what_ it does.
 For an in-depth, external resource about writing developer documentation, see [_Docs for Developers_](https://docsfordevelopers.com/).
 
-### Formatting code examples
+### Format code examples
 
-Here are guidelines to follow when formatting code examples.
+To format code examples, use the following guidelines.
 
-- Use spaces, not tabs.
-- Follow Grafana's accepted [coding style guidelines](https://github.com/grafana/grafana/blob/main/contribute/style-guides/).
-- Wrap lines at 80 characters.
-- When omitting code, use three dots (...). Don't use the ellipsis character (…).
+#### Highlight syntax
 
-Introduce each code sample with a sentence or paragraph to establish its context. End the introduction with a colon if it immediately precedes the sample, or a period if it doesn't.
+In Markdown, an info string after the first three backticks describes the language contained within.
+The website uses this information to apply syntax highlighting to code examples.
+
+The following Markdown sets the info string to `json`:
+
+````markdown
+```json
+{ "key": "value" }
+```
+````
+
+Some common languages and their info strings are:
+
+- Bash: `bash`
+- Console: `console`
+- Go: `go`
+- JSON: `json`
+- PromQL: `promql`
+- River: `river`
+- Shell: `shell`
+- YAML: `yaml`
+
+#### Use appropriate whitespace for indentation
+
+Spaces (` `) are a good default but note that some languages use alternative indentation.
+For example, Makefiles and Go source code uses tabs (`	`) for indentation.
+
+Use 2 spaces unless there is a clearly established alternative convention.
+For example, Python generally uses 4 spaces for indentation.
+
+Above all else, be consistent with other Grafana documentation.
```

Comment:
What exactly do we mean here? Like if you're adding something, just make sure it's consistent with the other code samples around it? My confusion comes from the phrase "other Grafana documentation" because I interpret that as ALL other Grafana documentation.

Reply:
I see Isabel's point. Is what we are trying to say "Above all else, be consistent with other Grafana documentation for code in the same language."?

Reply:
Perhaps it is more confusing that it is valuable. I'm not sure exactly what I meant now that I have come back to this PR after some time.

I think it's kind of a hand-wavy "If your project is doing four tabs of indentation (WILD!) for all other code samples in this language, follow that style" but also "If Grafana engineering has a practice of doing X, then also do X."

Since this is the style guide, perhaps we should be more insistent that you do Y and if you are doing X, you must switch to doing Y.

Reply:
I think I'd say be consistent with existing documentation _in that area_ of the docs or in your project. I think we just want to be specific about what we're saying to be consistent with.

Reply:
```suggestion
Above all else, be consistent with existing documentation in your area of the documentation or in your project.
```